core/except.cpp

```diff
@@ -305,5 +305,15 @@ void validExtent(const scipp::index size) {
     throw except::DimensionError("Dimension size cannot be negative.");
 }
 
+void binEdgeOverlaps(const VariableConstProxy &lhs,
+                     const VariableConstProxy &rhs, const Dim dim) {
```

Comment:
I do not understand the function name. Isn't this checking that there is *no* overlap (but also no gap)?

Reply:
Yes, maybe `binEdgeMatches` is a better name?

It's checking for an overlapping/matching bin edge, the fact it is edges not bins and edge is singular I assumed made it apparent is was only one.

Reply:
Still does not say what it does (could be comparing all bin edges?). Maybe `lastSliceEqualsFirstSlice`?

Reply:
`lastSliceEqualsFirstSlice` doesn't indicate how big the slice is.

`binsAreConsectiveButNotOverlapping`?

Reply:
How about you just define this helper locally, then the name does not matter so much? Unless it is generic enough to be useful there is no need to put it inti `except.h`.